src/libs/Navigation/Navigation.js

```diff
@@ -51,6 +52,9 @@ function dismissModal() {
     // From there we can just navigate back and open the drawer
     goBack();
     openDrawer();
+
+    // Store modal closed state in Onyx
+    setModalVisibility(false);
```

Comment:
Hmm there are other ways to dismiss the modal e.g. by swiping - which I think we are not covering with just this.

Reply:
Looks like you're right - on iOS:

> The `modal` behavior slides the screen in from the bottom on iOS and allows the user to swipe down from the top to dismiss it. The `modal` prop has no effect on Android because full-screen modals don't have any different transition behavior on the platform.

Source: https://reactnavigation.org/docs/modal/

Reply:
looks like `beforeRemove` is called when dismissed via `Navigation.dismissModal` OR when dismissed on iOS 👍🏽